packages/pug/lib/index.js

```diff
@@ -303,6 +305,7 @@ exports.compileClientWithDependenciesTracked = function(str, options){
     filterOptions: options.filterOptions,
     filterAliases: options.filterAliases,
     plugins: options.plugins,
+    module: options.module
   });
 
   return {body: parsed.body, dependencies: parsed.dependencies};
```

Comment:
Please make your changes **here**.  i.e. add some code like:

```js
var body = parsed.body;
if (options.body) {
  // wrap body in module here
}
return {body: body, dependencies: parsed.dependencies};
```

That way we don't make pug-code-gen any more complex.